### ISSUE TITLE
Error ignored in parse_pairs

### ISSUE THREAD

**Issue by tonis** [canaryhealth/pyramid_input#1]
Hi, first of all, it's an incredibly useful plugin for pyramid. It enables the posting of complex data structures that I'd otherwise have no way to feed to WTForms. Or had to do a lot of stuff manually, thank you!

One thing I encountered was this: I accidentally posted a malformed form and the had to guess why my POST data wasn't parsed as it should have been.

When you try something like this:

> aa = {'liige-1.name': 'name', 'liige.0.country': 'EE', 'liige-0.name': ''}
> formencode.variabledecode.variable_decode(aa) 

then it gives the exception, 

TypeError: unorderable types: str() < int()
there is a . too many in "liige.0.country", the first "." should have been a "-".

It would be nice if it logged an error, not ignored it silently. I could do it myself but I don't know how the error should be logged, for debugging purposes I just added a print() statement there but that won't be nice.

**Comment by grabner** [comment]
although i agree that there should be some feedback of the error and have just added that to 0.2.1 (changeset cbbc9b65a94ba268f89011be67acdfd14a76b0af), i was not able to reproduce your error:

``` python
>>> import formencode.variabledecode
>>> aa = {'liige-1.name': 'name', 'liige.0.country': 'EE', 'liige-0.name': ''}
>>> formencode.variabledecode.variable_decode(aa)
{'liige': [{'name': ''}, {'name': 'name'}, {'country': 'EE'}]}
```

can you send me the output of a `pip freeze` to see what package versions you are using?

**Comment by tonis** [comment]
Hello, my FormEncode version is 1.3.0

**Comment by grabner** [comment]
ah - this is a formencode+py3 issue (it "works" in py2 by appending the invalid key/value to the list as a new pair)... i've opened a ticket in formencode/formencode#92 for this and created a unit test here (a429d7b3d257ba2063985924b481945865d5a615).

in any case, this issue is now addressed in 0.2.1 (py2 gets odd output, py3 logs an exception).
